src/libs/viewpoints/viewpoint_extractor.ts

```diff
@@ -0,0 +1,536 @@
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(viewpoint.classes));
+
+    viewpoint.removeDuplicates();
+
+    return viewpoint;
+  }
+
+  /**
+     * This method filter all Relators, respective direct conectect clases by mediator relation, their sortal ancestors  
+     * and related relations and group them in a view
+     * 
+     * @returns A viewpoint extracted using the Relator Pattern
+     */
+  relatorView = () => {
+    const viewpoint = new Viewpoint('Relator View');
+
+
+    /**Step 1 Get all Relators in model
+     * 
+     * Step 2 Get all classes conected by mediation relation 
+     * 
+     * Step 3 Get all sortal ancestors 
+     * 
+     * Step 4 Get all relations between all classes 
+     * 
+     * Step 5 Get all generalizations between all classes 
+     */
+    let classes = this.project.getAllClassesByStereotype(ClassStereotype.RELATOR);
+
+    const relacoes = this.project.getAllRelations();
+    let mediacoes = [];
+
+    for (let j = 0; j < classes.length; j++) {
+      for (let i = 0; i < relacoes.length; i++) {
+        if ((this.isClassInRelation(classes[j], relacoes[i])) && (relacoes[i].hasMediationStereotype())) {
+          mediacoes = mediacoes.concat(relacoes[i]);
+        }
+      }
+    }
+
+    for (let i = 0; i < mediacoes.length; i++) {
+      if (!mediacoes[i].getSourceClass().hasRelatorStereotype()) {
+        classes = classes.concat(mediacoes[i].getSourceClass());
+      } else {
+        if (!mediacoes[i].getTargetClass().hasRelatorStereotype()) {
+          classes = classes.concat(mediacoes[i].getTargetClass());
+        }
+      }
+    }
+
+    viewpoint.addClasses(classes);
+
+    viewpoint.removeDuplicates();
+
+
+    viewpoint.addRelations(this.getRelationsBetweenClasses(viewpoint.classes));
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(viewpoint.classes));
+
+    viewpoint.removeDuplicates();
+
+    return viewpoint;
+  }
+
+  /**
+     * This method filter all Non-Sortals, their descendants, then the decendants sortal ancestors  and related 
+     * relations and group them in a view
+     * 
+     * @returns A viewpoint extracted using the Non-Sortal Pattern
+     */
+  nonSortalView(): Viewpoint {
+    const viewpoint = new Viewpoint('Non Sortal View');
+
+    /**Step 1 Get all Non-Sortals in model
+     * 
+     * Step 2 Get all non-sortals decendants 
+     * 
+     * Step 3 Get all decendants sortal ancestors 
+     * 
+     * Step 4 Get all relations between all classes 
+     * 
+     */
+    let classes = this.project.getAllClassesByStereotype(ClassStereotype.ROLE_MIXIN);
+
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.MIXIN));
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.CATEGORY));
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.PHASE_MIXIN));
```

Comment:
Please add a method to the `Project` class called `getNonSortals()` that returns classes with these 4 stereotypes.

Reply:
Done